source/extensions/tracers/skywalking/tracer.cc

```diff
@@ -77,6 +77,15 @@ void Tracer::sendSegment(TracingContextPtr segment_context) {
   }
 }
 
+std::string Span::getTraceIdAsHex() const {
+  auto sw_trace_id = tracing_context_->traceId();
+  if (sw_trace_id.size() == 36) {
+    return absl::StrReplaceAll(sw_trace_id, {{"-", ""}});
+  }
+
+  return EMPTY_STRING;
+}
```

Comment:
Could you explain why we do this in this way?

And does this make sense? The `getTraceIdAsHex` mainly is used for logging. But this replacement make we log a id different with the actually id. That means the users need to handle the id first to add the "-" back, then the id in log could relate to an actual trace chain.

Reply:
yes, this's a little odd. Share same logic with OpenTelemetry collector:

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/c233bc7b20850fb5febecbf38d6adae3616bc13e/pkg/translator/skywalking/skywalkingproto_to_traces.go#L265

Do you have a better suggestion?

Reply:
I am not sure. And I don't know how the end users benefit from it

Reply:
this will help people using skywalking tracer and OpenTelemetry accesslogger, maybe we can colse this first, and reopen when there's a real requirement, WDYT?

Reply:
> this will help people using skywalking tracer and OpenTelemetry accesslogger

Yeah. I agree with this point, but doubt the specific implementation. The users may hope a unique id to correlate the log entry and trace chain. But now, they must handle the id in the log entry and do a conversion first.

> maybe we can colse this first, and reopen when there's a real requirement, WDYT

SGTM.